waku/v2/waku_archive/driver/postgres_driver/postgres_driver.nim

```diff
@@ -234,41 +234,85 @@ method getMessages*(s: PostgresDriver,
-    return err("failed to get messages count: rows.len == 0")
+  if rows.len != 1:
+    return err("failed in getRow. Expected one row but got " & $rows.len)
+
+  let fields = rows[0]
+  if fields.len != 1:
+    return err("failed in getRow: Expected one field but got " & $fields.len)
 
-  let rowFields = rows[0]
-  if rowFields.len == 0:
-    return err("failed to get messages count: rowFields.len == 0")
+  var retInt: int64
+  try:
+    retInt = parseInt(fields[0])
+  except ValueError:
+    return err("exception in getRow, parseInt: " & getCurrentExceptionMsg())
+
+  return ok(retInt)
+
+method getMessagesCount*(s: PostgresDriver):
+                         Future[ArchiveDriverResult[int64]] {.async.} =
 
-  let count = parseInt(rowFields[0])
-  return ok(count)
+  let intRes = await s.getInt("SELECT COUNT(1) FROM messages")
+  if intRes.isErr():
+    return err("error in getMessagesCount: " & intRes.error)
+
+  return ok(intRes.get())
 
 method getOldestMessageTimestamp*(s: PostgresDriver):
                                   Future[ArchiveDriverResult[Timestamp]] {.async.} =
-  return err("not implemented")
+
+  let intRes = await s.getInt("SELECT MIN(storedAt) FROM messages")
+  if intRes.isErr():
+    return err("error in getOldestMessageTimestamp: " & intRes.error)
+
+  return ok(Timestamp(intRes.get()))
 
 method getNewestMessageTimestamp*(s: PostgresDriver):
                                   Future[ArchiveDriverResult[Timestamp]] {.async.} =
-  return err("not implemented")
 
-method deleteMessagesOlderThanTimestamp*(s: PostgresDriver,
-                                         ts: Timestamp):
-                                         Future[ArchiveDriverResult[void]] {.async.} =
-  return err("not implemented")
+  let intRes = await s.getInt("SELECT MAX(storedAt) FROM messages")
+  if intRes.isErr():
+    return err("error in getOldestMessageTimestamp: " & intRes.error)
+
+  return ok(Timestamp(intRes.get()))
+
+method deleteMessagesOlderThanTimestamp*(
+                                 s: PostgresDriver,
+                                 ts: Timestamp):
+                                 Future[ArchiveDriverResult[void]] {.async.} =
+
+  let execRes = await s.connPool.exec(
+                            "DELETE FROM messages WHERE storedAt < " & $ts)
+  if execRes.isErr():
+    return err("error in deleteMessagesOlderThanTimestamp: " & execRes.error)
+
+  return ok()
 
-method deleteOldestMessagesNotWithinLimit*(s: PostgresDriver,
-                                           limit: int):
-                                           Future[ArchiveDriverResult[void]] {.async.} =
-  return err("not implemented")
+method deleteOldestMessagesNotWithinLimit*(
+                                 s: PostgresDriver,
+                                 limit: int):
+                                 Future[ArchiveDriverResult[void]] {.async.} =
+
+  let execRes = await s.connPool.exec(
+                     """DELETE FROM messages WHERE id NOT IN
```

Comment:
Probably not a problem, but this could potentially result in large set of `IN`, not sure if there could be some performance penalty when a lot of messages (thousands?) would need to be deleted. On the other hand, my SQL-fu is weak, so not sure if there is another way:)